pkg/cloud/services/networking/client.go

```diff
@@ -163,21 +171,55 @@ func (c networkClient) ListPort(opts ports.ListOptsBuilder) ([]ports.Port, error
 }
 
 func (c networkClient) CreatePort(opts ports.CreateOptsBuilder) (*ports.Port, error) {
-	return ports.Create(c.serviceClient, opts).Extract()
+	mc := metrics.NewMetricPrometheusContext("port", "create")
+	port, err := ports.Create(c.serviceClient, opts).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return port, nil
 }
 
 func (c networkClient) DeletePort(id string) error {
-	return ports.Delete(c.serviceClient, id).ExtractErr()
+	mc := metrics.NewMetricPrometheusContext("port", "delete")
+	return mc.ObserveRequest(ports.Delete(c.serviceClient, id).ExtractErr())
 }
 
 func (c networkClient) GetPort(id string) (*ports.Port, error) {
-	return ports.Get(c.serviceClient, id).Extract()
+	mc := metrics.NewMetricPrometheusContext("port", "get")
+	port, err := ports.Get(c.serviceClient, id).Extract()
+	if mc.ObserveRequestIgnoreNotFound(err) != nil {
+		return nil, err
+	}
+	return port, nil
 }
 
 func (c networkClient) UpdatePort(id string, opts ports.UpdateOptsBuilder) (*ports.Port, error) {
 	return ports.Update(c.serviceClient, id, opts).Extract()
 }
 
+func (c networkClient) CreateTrunk(opts trunks.CreateOptsBuilder) (*trunks.Trunk, error) {
+	mc := metrics.NewMetricPrometheusContext("trunk", "create")
+	trunk, err := trunks.Create(c.serviceClient, opts).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return trunk, nil
+}
+
+func (c networkClient) DeleteTrunk(id string) error {
+	mc := metrics.NewMetricPrometheusContext("trunk", "delete")
+	return mc.ObserveRequest(trunks.Delete(c.serviceClient, id).ExtractErr())
+}
+
+func (c networkClient) ListTrunk(opts trunks.ListOptsBuilder) ([]trunks.Trunk, error) {
```

Comment:
Bikeshed: This should be plural: ListTrunks (along with all the other List* methods).

Leave it for now unless you want to change all the others in a separate commit, and it's obviously not super important! It does make me wince, though 😉